lib/main.dart

```diff
@@ -13,50 +16,92 @@ class MyApp extends StatelessWidget {
-        }
-        return _buildRow(_suggestions[index]);
-      });
-}
-  
-  @override
-    Widget build(BuildContext context) {
-      return Scaffold(
-    appBar: AppBar(
-      title: const Text('Startup Name Generator'),
-    ),
-    body: _buildSuggestions(),
-  );
-    }
+    return ListView.builder(
+        padding: const EdgeInsets.all(16.0),
+        itemBuilder: /*1*/ (context, i) {
+          if (i.isOdd) return const Divider(); /*2*/
 
+          final index = i ~/ 2; /*3*/
+          if (index >= _suggestions.length) {
+            _suggestions.addAll(generateWordPairs().take(10)); /*4*/
+          }
+          return _buildRow(_suggestions[index]);
+        });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final punchLines = [
+      "My girlfriend told me to stop making linkin park references. But in the end, it didn't even matter.",
+      "I can't stand being in a wheelchair.",
+      "A woman walks into a library and asked if they had any books about paranoia. Librarian: They're right behind you!!",
+      "I know a lot of jokes about unemployed people but none of them work.",
+      "Why do blind people hate skydiving? It scares the hell out of their dogs.",
+      "At least this wasn't a blind joke. I can't see how those are funny.",
+      "Never trust an atom. They make up everything.",
+      "How can you spot a blind man at a nude beach? It isn't hard.",
+      "It's hard to explain puns to kleptomaniacs because they always take things literally.",
+      "Communism jokes aren't funny unless everyone gets them",
+      "Why don't ants get sick? Because they have little antybodies.",
+      "Shout out to the people who want to know what the opposite of in is",
+      "A blind man walks into a bar. And a table. And a chair.",
+      "A dyslexic walks into a bra.",
+      "Hey do you want to hear a joke about construction? nevermind i'm still working on it",
+      "I used to be addicted to soap, but now I'm clean...",
+      "I poured root beer in a square glass. Now I just have beer.",
+      "My girlfriend asked if I would stop listening to Oasis, I said maybe...",
+      "Say what you want about deaf people."
+    ];
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Neverending story: flutter edition'),
+      ),
+      body: _buildSuggestions(),
+      floatingActionButton: new FloatingActionButton(
+          tooltip: 'Add',
+          child: new Icon(Icons.cake_outlined),
+          onPressed: () {
+            showDialog(
+              context: context,
+              builder: (BuildContext context) {
+                return AlertDialog(
+                  title: const Text("Nice!"),
+                  content: new Text("You've been rewarded with a punchline. Click the button to claim your prize"),
+                  actions: <Widget>[
+                    new TextButton(
+                      child: new Text("CLAIM"),
```

Comment:
AFAIK you can omit the `new` keyword; it became optional with dart 2.0+

Reply:
Thanks a lot! Will look into it;)